src/stan/lang/ast/fun/var_occurs_vis_def.hpp

```diff
@@ -62,6 +62,14 @@ namespace stan {
       return false;  // no refs persist out of integrate_ode_control() call
     }
 
+    bool var_occurs_vis::operator()(const algebra_solver& e) const {
+      return false;
```

Comment:
I'm not sure. I followed the example for the ode integrator here. Judging by the comment:

```
// no refs persist out of integrate_ode_control() call
```
I would say this only matters if one of the argument is passed by ref and modified. Adding the comments for the algebraic solver.